Release 2.8.1 (#4773)
timescale/timescaledb: CHANGELOG.md

**Bugfixes**

> * #4619 Improve handling enum columns in compressed hypertables

Reviewer: ```suggestion
* #4619 Improve handling enum columns in compressed hypertables
```
This references the issue and not the PR. #4658 is the correct reference which is part of 2.8.1

Author: Good catch, entry is removed from this section

* #4673 Fix now() constification for VIEWs
* #4681 Fix compression_chunk_size primary key
* #4696 Report warning when enabling compression on hypertable
* #4745 Fix FK constraint violation error while insert into hypertable which references partitioned table
* #4756 Improve compression job IO performance

**Thanks**

> * @jvanns for reporting hypertable FK reference to vanilla PostgreSQL partitioned table doesn't seem to work

Reviewer: ```suggestion
```
This needs to go into 2.8.1 section since it belongs to #2658

Author: @svenklemm This belongs to #4745, which is not included so far (the PR was merged after I created the initial list of commits that should be included in the release). Are you in favor of including the #4745 PR also in the 2.8.1?

## 2.8.1 (2022-10-04)

This release is a patch release. We recommend that you upgrade at the
next available opportunity.

**Bugfixes**
* #4454 Keep locks after reading job status
* #4658 Fix error when querying a compressed hypertable with compress_segmentby on an enum column
* #4671 Fix a possible error while flushing the COPY data
* #4675 Fix bad TupleTableSlot drop
* #4676 Fix a deadlock when decompressing chunks and performing SELECTs
* #4685 Fix chunk exclusion for space partitions in SELECT FOR UPDATE queries
* #4694 Change parameter names of cagg_migrate procedure
* #4698 Do not use row-by-row fetcher for parameterized plans
* #4711 Remove support for procedures as custom checks
* #4712 Fix assertion failure in constify_now
* #4713 Fix Continuous Aggregate migration policies
* #4720 Fix chunk exclusion for prepared statements and dst changes
* #4726 Fix gapfill function signature
* #4737 Fix join on time column of compressed chunk
* #4738 Fix error when waiting for remote COPY to finish
* #4739 Fix continuous aggregate migrate check constraint
* #4760 Fix segfault when INNER JOINing hypertables

> * #4767 Allow manual index creation for CAggs

Reviewer 2: ```suggestion
* #4767 Fix permission issues on index creation for CAggs
```

I guess the current explanation is not very good because seems we're now allowing index creation on CAggs and actually it is a feature already released in 2.8.0 and this PR just fix a permission issue in this new feature.

Author: 👍  Is changed in the current version of the PR

* @carobme for reporting constraint error during continuous aggregate migration
* @jvanns for reporting hypertable FK reference to vanilla PostgreSQL partitioned table doesn't seem to work

## 2.8.1 (2022-10-04)

This release is a patch release. We recommend that you upgrade at the
next available opportunity.

**Bugfixes**
* #4454 Keep locks after reading job status
* #4671 Fix a possible error while flushing the COPY data
* #4676 Fix a deadlock when decompressing chunks and performing SELECTs
* #4694 Change parameter names of cagg_migrate procedure
* #4698 Do not use row-by-row fetcher for parameterized plans
* #4711 Remove support for procedures as custom checks
* #4713 Fix Continuous Aggregate migration policies
* #4738 Fix error when waiting for remote COPY to finish
* #4739 Fix continuous aggregate migrate check constraint

> * #4686 Add CAGG migration permission tests

Reviewer: ```suggestion
```
We dont mention test-only changes in CHANGELOG

> ## 2.8.1 (2022-10-04)
> 
> This release is a patch release. We recommend that you upgrade at the
> next available opportunity.
> 
> **Bugfixes**
> * #4454 Keep locks after reading job status
> * #4671 Fix a possible error while flushing the COPY data
> * #4676 Fix a deadlock when decompressing chunks and performing SELECTs
> * #4694 Change parameter names of cagg_migrate procedure
> * #4698 Do not use row-by-row fetcher for parameterized plans
> * #4711 Remove support for procedures as custom checks
> * #4713 Fix Continuous Aggregate migration policies
> * #4738 Fix error when waiting for remote COPY to finish
> * #4739 Fix continuous aggregate migrate check constraint
> * #4686 Add CAGG migration permission tests
> 
> **Thanks**
> * @daniel-k for reporting a background worker crash
> 

Reviewer: ```suggestion
## 2.8.1 (2022-10-04)

This release is a patch release. We recommend that you upgrade at the
next available opportunity.

**Bugfixes**
* #4454 Keep locks after reading job status
* #4658 Fix error when querying a compressed hypertable with compress_segmentby on an enum column
* #4671 Fix a possible error while flushing the COPY data
* #4675 Fix bad TupleTableSlot drop
* #4676 Fix a deadlock when decompressing chunks and performing SELECTs
* #4685 Fix chunk exclusion for space partitions in SELECT FOR UPDATE queries
* #4694 Change parameter names of cagg_migrate procedure
* #4698 Do not use row-by-row fetcher for parameterized plans
* #4711 Remove support for procedures as custom checks
* #4712 Fix assertion failure in constify_now
* #4713 Fix Continuous Aggregate migration policies
* #4720 Fix chunk exclusion for prepared statements and dst changes
* #4726 Fix gapfill function signature
* #4733 Fix an ereport syntax error when using PG 12.0
* #4737 Fix join on time column of compressed chunk
* #4738 Fix error when waiting for remote COPY to finish
* #4739 Fix continuous aggregate migrate check constraint
* #4760 Fix segfault when INNER JOINing hypertables

**Thanks**
* @boxhock and @cocowalla for reporting a segfault when JOINing hypertables
* @carobme for reporting constraint error during continuous aggregate migration
* @choisnetm, @dustinsorensen, @jayadevanm and @joeyberkovitz for reporting a problem with JOINs on compressed hypertables
* @daniel-k for reporting a background worker crash
* @justinpryzby for reporting an error when compressing very wide tables
* @maxtwardowski for reporting problems with chunk exclusion and space partitions
* @yuezhihan for reporting GROUP BY error when having compress_segmentby on an enum column
```

Reviewer: Also any entry moved into 2.8.1 section should be removed from the unreleased section

Author: #4733 Fix an ereport syntax error when using PG 12.0 - This line was removed from the list, because the problem was introduced in the development version and was never released.

Author: The rest is updated in the current version of the PR.
